### ISSUE TITLE
docker / openstack cinder : unable to find files in volume

### ISSUE THREAD

**Issue by olivibernard** [codedellemc/rexray#1078]
# Summary
I configured rexray on 2 openstack nodes (manualy, I do not use Vagrant).
I launch the demo : http://rexray.readthedocs.io/en/stable/user-guide/examples/demo/#node-0
I can not get the persistence.
Wrong config ?

## Version
```shell
$ rexray version
REX-Ray
-------
Binary: /usr/bin/rexray
Flavor: client+agent+controller
SemVer: 0.11.0
OsArch: Linux-x86_64
Commit: 508023f4b0e4974f1844c26b39af7c4219f4c7a9
Formed: Mon, 16 Oct 2017 09:07:06 CEST

# docker version
Client:
 Version:      17.09.0-ce
 API version:  1.32
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:42:18 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.09.0-ce
 API version:  1.32 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:40:56 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

## Expected Behavior
I would like to see the file(s) created from node0.

## Actual Behavior
I can not see the file(s) created from node0.

## Steps To Reproduce
On node0 :
```shell
root@node0:~# docker volume create --driver rexray --opt size=1 \
>                  --name hellopersistence
hellopersistence
root@node0:~# docker run -tid --volume-driver=rexray \
>                  -v hellopersistence:/mystore \
>                  --name temp01 busybox
Unable to find image 'busybox:latest' locally
latest: Pulling from library/busybox
03b1be98f3f9: Pull complete
Digest: sha256:3e8fa85ddfef1af9ca85a5cfb714148956984e02f00bec3f7f49d3925a91e0e7
Status: Downloaded newer image for busybox:latest
090838ad26048f3e92deaba5713eea73067eef5b7597532be62c0b7cfe313f35

root@node0:~#
root@node0:~# docker exec temp01 touch /mystore/myfile
root@node0:~# docker exec temp01 ls -l /mystore/myfile
-rw-r--r--    1 root     root             0 Oct 16 14:29 /mystore/myfile
root@node0:~# docker rm -f temp01
```

On node1 :
```shell
ubuntu@node1:~$ docker run -tid --volume-driver=rexray \
>                  -v hellopersistence:/mystore \
>                  --name temp01 busybox
Unable to find image 'busybox:latest' locally
latest: Pulling from library/busybox
03b1be98f3f9: Pull complete
Digest: sha256:3e8fa85ddfef1af9ca85a5cfb714148956984e02f00bec3f7f49d3925a91e0e7
Status: Downloaded newer image for busybox:latest
63e9d12c087327835368a35087f22146ff9394ed2bca82db0ee554082da1a336
ubuntu@node1:~$ docker exec temp01 ls -l /mystore
total 20
drwx------    2 root     root          4096 Oct 16 14:31 data
drwx------    2 root     root         16384 Oct 16 14:31 lost+found
```

## Configuration Files

```yaml
root@node0:/etc/rexray# cat config.yml
libstorage:
  integration:
    volume:
      operations:
        remove:
          force: true
          disable: true
        unmount:
          ignoreusedcount: true
        mount:
          preempt: true
          rootPath: "/"
          retrywait: "40s"
        create:
          default:
            size: 1
  service: cinder
  logging:
    level:         debug
    httpRequests:  true
    httpResponses: true
  server:
    services:
      cinder:
        driver: cinder
cinder:
  authUrl:              "https://mycompany.com:5000/v3"
  authVersion:          "3"
  identityApiVersion:   "2"
  username:             "xxxxx"
  tenantId:             "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
  password:             "xxxxxxxxxxx"
  domainName:           "xxxxxxxxxx"
  cacert:                "/ca-certificates.crt"
```
Any help will be appreciated, 
Thanks.

**Comment by clintkitson** [comment]
It looks like your configuration may be different on the second node.
Notice the rootPath line where you are specifying /.

**Comment by olivibernard** [comment]
@clintkitson 
Both config.yml files are the same on node0 and node1.
I removed the rootPath line, restarted rexray service, and run again.

On node0 : 
```
# docker volume create --driver rexray --opt size=1 --name hello
hello
# docker run -tid --volume-driver=rexray -v hello:/mystore --name temp02 busybox
3330304777b8cc8ded89c3e8a86033d1a7deaf3f628fb840e25a94db4aae7058
# docker ps
CONTAINER ID        IMAGE                  COMMAND                  CREATED             STATUS              PORTS               NAMES
3330304777b8        busybox                "sh"                     18 seconds ago      Up 8 seconds                            temp02
# docker exec temp02 touch /mystore/myfile
# docker exec temp02 ls -l /mystore
total 0
-rw-r--r--    1 root     root             0 Oct 17 06:51 myfile
# docker rm -f temp02
temp02
```
On node1, first attempt to mount : I get an error : 
```shell
# docker run -tid --volume-driver=rexray -v hello:/mystore --name temp02 busybox
ff599d2bbd3bd9bb6499f32fa81566331ea080abe9cde01cd0e88b91278b2890
docker: Error response from daemon: error while mounting volume '/': VolumeDriver.Mount: NodePublishVolume failed: 6: device not found: timed out.
root@yvao9990:/var/log/rexray# cat rexray.log
```
In rexray.log, I can see a HTTP request : 
```
time="2017-10-17T08:53:17+02:00" level=info msg="    -------------------------- HTTP REQUEST (CLIENT) -------------------------"
time="2017-10-17T08:53:17+02:00" level=info msg="    GET /volumes/cinder?attachments=0 HTTP/1.1"
time="2017-10-17T08:53:17+02:00" level=info msg="    Host: libstorage-server"
time="2017-10-17T08:53:17+02:00" level=info msg="    Libstorage-Instanceid: cinder=191c2dad-fba9-4137-b8d0-15d566a5066e"
time="2017-10-17T08:53:17+02:00" level=info msg="    Libstorage-Localdevices: cinder=/dev/dm-0::,/dev/dm-1::,/dev/vda::,/dev/vda1::,/dev/vdb::,/dev/vdc::,/dev/vdd::"
time="2017-10-17T08:53:17+02:00" level=info msg="    Libstorage-Tx: txID=b2eec3b2-a6bd-48a8-6a6b-016726866746, txCR=1508223197"
time="2017-10-17T08:53:17+02:00" level=info msg="    "
```
The instanceid corresponds to node1, where I tried to run the container.
In the HTTP response, I can not see the hello volume.
docker rm -f temp02
And then try again docker run : 
Now I can see the 'hello' volume in the HTTP response to GET /volumes/cinder?attachments=0
```shell
# docker run -tid --volume-driver=rexray -v hello:/mystore --name temp02 busybox
a01eb5ec19dc42ecb3cf900ee80334a4f93c07a6e676f88b87c855565a0c6b06
docker: Error response from daemon: error while mounting volume '/var/lib/rexray/docker/volumes/hello': VolumeDriver.Mount: ControllerPublishVolume failed: 0: ControllerPublishVolume failed: 0: rpc error: code = Unknown desc = unable to find attached vol in local devices.
```

**Comment by olivibernard** [comment]
after more attempts, the container is running on node1, but the filesystem /mystore is empty : 
```shell
root@node1:/var/log/rexray# docker rm -f temp02
temp02
root@node1:/var/log/rexray# docker run -tid --volume-driver=rexray -v hello:/mystore --name temp02 busybox
288f2c1851295269948d618e1ea3df11599a321987bc69141aa1d650ebb12083
root@node1:/var/log/rexray# docker ps
CONTAINER ID        IMAGE                  COMMAND                  CREATED             STATUS              PORTS               NAMES
288f2c185129        busybox                "sh"                     3 minutes ago       Up 2 minutes                            temp02

root@node1:/var/log/rexray# docker exec temp02 ls -l /mystore
total 0

**Comment by clintkitson** [comment]
@olivibernard Can you use the `rexray volume mount hello` command on each node to test this again? Once it is mounted please review the filesystem that was attached and what files exist or don't exist.

**Comment by olivibernard** [comment]
@clintkitson I see a weird behaviour after mounting on node0 : 
```
# mount |grep rexray
# 
# df -h |grep rexray
#
# rexray volume ls
ID                                    Name                       Status       Size
ec41dcbb-9362-49ff-9650-7ab2e3ce7e5b  hello                      available    1

# rexray volume attach hello
ID                                    Name   Status    Size
ec41dcbb-9362-49ff-9650-7ab2e3ce7e5b  hello  attached  1

# rexray volume mount hello
INFO[0000] updated log level                             logLevel=debug
DEBU[0000] created scoped scope                          new=rexray.cli parentScopes=rexray,
DEBU[0000] os.args                                       time=1508406545341 val=[rexray volume mount hello]
DEBU[0000] activating libStorage                         cmd=mount time=1508406545341
DEBU[0000] parseSafeHost - no change                     postParse=unix:///var/run/rexray/062151526.sock preParse=unix:///var/run/rexray/062151526.sock time=1508406545                              349
DEBU[0000] read spec file                                host=unix:///var/run/rexray/062151526.sock path=/var/run/rexray/rexray.spec time=1508406545349
DEBU[0000] is local unix server active                   sock=/var/run/rexray/062151526.sock time=1508406545349
DEBU[0000] set host in context                           host=unix:///var/run/rexray/062151526.sock time=1508406545349
DEBU[0000] set host in config                            host=unix:///var/run/rexray/062151526.sock time=1508406545349
DEBU[0000] not starting embedded server; already running  host=unix:///var/run/rexray/062151526.sock time=1508406545350
DEBU[0000] creating libStorage client                    cmd=mount host=unix:///var/run/rexray/062151526.sock time=1508406545350
DEBU[0000] created scoped scope                          new=libstorage.client parentScopes=rexray.cli,rexray,
INFO[0000] configured logging                            host=unix:///var/run/rexray/062151526.sock libstorage.logging.httpRequests=true libstorage.logging.httpRespons                              es=true libstorage.logging.level=debug time=1508406545472
INFO[0000] set client service name                       host=unix:///var/run/rexray/062151526.sock service=cinder serviceName=cinder time=1508406545478
DEBU[0000] got configured host address                   host=unix:///var/run/rexray/062151526.sock service=cinder storageDriver=libstorage time=1508406545484
DEBU[0000] disabling tls for unix sockets                host=unix:///var/run/rexray/062151526.sock service=cinder storageDriver=libstorage time=1508406545488
DEBU[0000] getHost proto == "unix"                       getHost=libstorage-server host=unix:///var/run/rexray/062151526.sock service=cinder storageDriver=libstorage t                              ime=1508406545488
INFO[0000] created libStorage client                     clientType=integration disableKeepAlive=false enableInstanceIDHeaders=true enableLocalDevicesHeaders=true host                              =unix:///var/run/rexray/062151526.sock lAddr=libstorage-server logRequests=true logResponses=true lsxPath= service=cinder storageDriver=libstorage time=1508406545513
INFO[0000] registered custom context key                 externalID=Libstorage-Tx internalID=2 keyBitmask=2
INFO[0000] registered custom context key                 externalID=Libstorage-Instanceid internalID=3 keyBitmask=2
INFO[0000] registered custom context key                 externalID=Libstorage-Localdevices internalID=4 keyBitmask=2
INFO[0000] registered custom context key                 externalID=Authorization internalID=5 keyBitmask=2
INFO[0000]
INFO[0000]     -------------------------- HTTP REQUEST (CLIENT) -------------------------
INFO[0000]     GET /services HTTP/1.1
INFO[0000]     Host: libstorage-server
INFO[0000]     Libstorage-Tx: txID=ab042598-b26c-4f30-7210-7de77201ba90, txCR=1508406545
INFO[0000]
DEBU[0000] successful connection                         host=unix:///var/run/rexray/062151526.sock service=cinder storageDriver=libstorage time=1508406545514
INFO[0000]
INFO[0000]     -------------------------- HTTP RESPONSE (CLIENT) -------------------------
INFO[0000]     HTTP/1.1 200 OK
INFO[0000]     Content-Length: 111
INFO[0000]     Content-Type: application/json
INFO[0000]     Date: Thu, 19 Oct 2017 09:49:05 GMT
INFO[0000]     Libstorage-Servername: helix-cockatoo-az
INFO[0000]
INFO[0000]     {
INFO[0000]       "cinder": {
INFO[0000]         "name": "cinder",
INFO[0000]         "driver": {
INFO[0000]           "name": "cinder",
INFO[0000]           "type": "block"
INFO[0000]         }
INFO[0000]       }
INFO[0000]     }
INFO[0000] initializing supported cache                  host=unix:///var/run/rexray/062151526.sock server=helix-cockatoo-az service=cinder storageDriver=libstorage ti                              me=1508406545518
DEBU[0000] cached supported flag                         host=unix:///var/run/rexray/062151526.sock server=helix-cockatoo-az service=cinder storageDriver=libstorage su                              pported=15 time=1508406545518 txCR=1508406545 txID=83ed03f0-a145-4e92-62d1-cc45dda56cd2
INFO[0000] initializing instance ID cache                host=unix:///var/run/rexray/062151526.sock server=helix-cockatoo-az service=cinder storageDriver=libstorage ti                              me=1508406545519
DEBU[0005] cached instanceID                             host=unix:///var/run/rexray/062151526.sock instanceID=cinder=49a7ddd2-4b78-4ad8-9c5d-93701db4f506 server=helix                              -cockatoo-az service=cinder storageDriver=libstorage time=1508406550534 txCR=1508406545 txID=f2030636-c8d0-44ce-57b8-b940384775f5
DEBU[0005] xli instanceID success                        host=unix:///var/run/rexray/062151526.sock instanceID=cinder=49a7ddd2-4b78-4ad8-9c5d-93701db4f506 server=helix                              -cockatoo-az service=cinder storageDriver=libstorage time=1508406550534 txCR=1508406545 txID=f2030636-c8d0-44ce-57b8-b940384775f5
INFO[0005] successefully dialed libStorage server        host=unix:///var/run/rexray/062151526.sock server=helix-cockatoo-az service=cinder storageDriver=libstorage ti                              me=1508406550534
INFO[0005] storage driver initialized                    host=unix:///var/run/rexray/062151526.sock service=cinder storageDriver=libstorage time=1508406550534
INFO[0005] os driver initialized                         host=unix:///var/run/rexray/062151526.sock osDriver=linux service=cinder storageDriver=libstorage time=1508406                              550540
INFO[0005] linux integration driver successfully initialized  host=unix:///var/run/rexray/062151526.sock integrationDriver=linux libstorage.integration.volume.operatio                              ns.create.default.IOPS=0 libstorage.integration.volume.operations.create.default.availabilityZone= libstorage.integration.volume.operations.create.default.fsType=ext4                               libstorage.integration.volume.operations.create.default.size=1 libstorage.integration.volume.operations.create.default.type= libstorage.integration.volume.operations.c                              reate.implicit=true libstorage.integration.volume.operations.mount.path=/var/lib/rexray/volumes libstorage.integration.volume.operations.mount.rootPath=/data osDriver=                              linux service=cinder storageDriver=libstorage time=1508406550578
INFO[0005] path cache initializion disabled              host=unix:///var/run/rexray/062151526.sock integrationDriver=linux osDriver=linux service=cinder storageDriver                              =libstorage time=1508406550589
INFO[0005] libStorage integration driver successfully initialized  host=unix:///var/run/rexray/062151526.sock integrationDriver=linux libstorage.integration.volume.ope                              rations.create.disable=false libstorage.integration.volume.operations.mount.preempt=true libstorage.integration.volume.operations.path.cache.async=true libstorage.inte                              gration.volume.operations.path.cache.enabled=false libstorage.integration.volume.operations.remove.disable=true libstorage.integration.volume.operations.unmount.ignore                              usedcount=false osDriver=linux service=cinder storageDriver=libstorage time=1508406550614
INFO[0005] integration driver initialized                host=unix:///var/run/rexray/062151526.sock integrationDriver=linux osDriver=linux service=cinder storageDriver                              =libstorage time=1508406550615
INFO[0005] created libStorage client                     host=unix:///var/run/rexray/062151526.sock integrationDriver=linux osDriver=linux service=cinder storageDriver                              =libstorage time=1508406550615
DEBU[0005] xli localdevices success                      host=unix:///var/run/rexray/062151526.sock instanceID=cinder=49a7ddd2-4b78-4ad8-9c5d-93701db4f506 integrationD                              river=linux osDriver=linux server=helix-cockatoo-az service=cinder storageDriver=libstorage time=1508406550615 txCR=1508406550 txID=a8631ca5-931a-4d98-694c-2c389f2bc68                              3
INFO[0005]
INFO[0005]     -------------------------- HTTP REQUEST (CLIENT) -------------------------
INFO[0005]     GET /volumes/cinder?attachments=27 HTTP/1.1
INFO[0005]     Host: libstorage-server
INFO[0005]     Libstorage-Instanceid: cinder=49a7ddd2-4b78-4ad8-9c5d-93701db4f506
INFO[0005]     Libstorage-Localdevices: cinder=/dev/dm-0::,/dev/dm-1::,/dev/vda::,/dev/vda1::,/dev/vdb::,/dev/vdc::,/dev/vdf::
INFO[0005]     Libstorage-Tx: txID=a8631ca5-931a-4d98-694c-2c389f2bc683, txCR=1508406550
INFO[0005]
INFO[0006]
INFO[0006]     -------------------------- HTTP RESPONSE (CLIENT) -------------------------
INFO[0006]     HTTP/1.1 200 OK
INFO[0006]     Transfer-Encoding: chunked
INFO[0006]     Content-Type: application/json
INFO[0006]     Date: Thu, 19 Oct 2017 09:49:11 GMT
INFO[0006]     Libstorage-Servername: helix-cockatoo-az
INFO[0006]
INFO[0006]     b81
INFO[0006]     {
INFO[0006]       "1ed07588-3472-4a21-ac14-1a67709f5582": {
INFO[0006]         "attachments": [
INFO[0006]           {
INFO[0006]             "deviceName": "/dev/vda",
INFO[0006]             "instanceID": {
INFO[0006]               "id": "49a7ddd2-4b78-4ad8-9c5d-93701db4f506",
INFO[0006]               "driver": "cinder",
INFO[0006]               "service": ""
INFO[0006]             },
INFO[0006]             "status": "",
INFO[0006]             "volumeID": "1ed07588-3472-4a21-ac14-1a67709f5582"
INFO[0006]           }
INFO[0006]         ],
INFO[0006]         "attachmentState": 2,
INFO[0006]         "availabilityZone": "nova",
INFO[0006]         "name": "node0-boot",
INFO[0006]         "size": 20,
INFO[0006]         "status": "in-use",
INFO[0006]         "id": "1ed07588-3472-4a21-ac14-1a67709f5582",
INFO[0006]         "type": "svc-s21"
INFO[0006]       },
INFO[0006]       "3cd9680a-65ce-4bd8-a20b-e5d17439742a": {
INFO[0006]         "attachments": [
INFO[0006]           {
INFO[0006]             "deviceName": "/dev/vdc",
INFO[0006]             "instanceID": {
INFO[0006]               "id": "49a7ddd2-4b78-4ad8-9c5d-93701db4f506",
INFO[0006]               "driver": "cinder",
INFO[0006]               "service": ""
INFO[0006]             },
INFO[0006]             "status": "",
INFO[0006]             "volumeID": "3cd9680a-65ce-4bd8-a20b-e5d17439742a"
INFO[0006]           }
INFO[0006]         ],
INFO[0006]         "attachmentState": 2,
INFO[0006]         "availabilityZone": "nova",
INFO[0006]         "name": "node0-var-lib-docker",
INFO[0006]         "size": 70,
INFO[0006]         "status": "in-use",
INFO[0006]         "id": "3cd9680a-65ce-4bd8-a20b-e5d17439742a",
INFO[0006]         "type": "svc-s21"
INFO[0006]       },
INFO[0006]       "98d91360-4310-49f4-8c7d-15f6b5e450d0": {
INFO[0006]         "attachmentState": 3,
INFO[0006]         "availabilityZone": "nova",
INFO[0006]         "name": "yvao4270-users",
INFO[0006]         "size": 30,
INFO[0006]         "status": "available",
INFO[0006]         "id": "98d91360-4310-49f4-8c7d-15f6b5e450d0",
INFO[0006]         "type": "svc-s21"
INFO[0006]       },
INFO[0006]       "b1c1663f-2eb5-4a87-86e8-bf47a9a08f1f": {
INFO[0006]         "attachmentState": 3,
INFO[0006]         "availabilityZone": "nova",
INFO[0006]         "name": "",
INFO[0006]         "size": 5,
INFO[0006]         "status": "available",
INFO[0006]         "id": "b1c1663f-2eb5-4a87-86e8-bf47a9a08f1f",
INFO[0006]         "type": "svc-s01"
INFO[0006]       },
INFO[0006]       "b9603f10-c5d5-4b0a-a9a0-47936be39074": {
INFO[0006]         "attachments": [
INFO[0006]           {
INFO[0006]             "deviceName": "/dev/vdb",
INFO[0006]             "instanceID": {
INFO[0006]               "id": "49a7ddd2-4b78-4ad8-9c5d-93701db4f506",
INFO[0006]               "driver": "cinder",
INFO[0006]               "service": ""
INFO[0006]             },
INFO[0006]             "status": "",
INFO[0006]             "volumeID": "b9603f10-c5d5-4b0a-a9a0-47936be39074"
INFO[0006]           }
INFO[0006]         ],
INFO[0006]         "attachmentState": 2,
INFO[0006]         "availabilityZone": "nova",
INFO[0006]         "name": "node0-users",
INFO[0006]         "size": 30,
INFO[0006]         "status": "in-use",
INFO[0006]         "id": "b9603f10-c5d5-4b0a-a9a0-47936be39074",
INFO[0006]         "type": "svc-s21"
INFO[0006]       },
INFO[0006]       "db5d7770-2f62-4400-9808-0554c48c446f": {
INFO[0006]         "attachmentState": 3,
INFO[0006]         "availabilityZone": "nova",
INFO[0006]         "name": "yvao4270-boot",
INFO[0006]         "size": 20,
INFO[0006]         "status": "available",
INFO[0006]         "id": "db5d7770-2f62-4400-9808-0554c48c446f",
INFO[0006]         "type": "svc-s21"
INFO[0006]       },
INFO[0006]       "ec41dcbb-9362-49ff-9650-7ab2e3ce7e5b": {
INFO[0006]         "attachments": [
INFO[0006]           {
INFO[0006]             "deviceName": "/dev/vdd",
INFO[0006]             "instanceID": {
INFO[0006]               "id": "49a7ddd2-4b78-4ad8-9c5d-93701db4f506",
INFO[0006]               "driver": "cinder",
INFO[0006]               "service": ""
INFO[0006]             },
INFO[0006]             "status": "",
INFO[0006]             "volumeID": "ec41dcbb-9362-49ff-9650-7ab2e3ce7e5b"
INFO[0006]           }
INFO[0006]         ],
INFO[0006]         "attachmentState": 2,
INFO[0006]         "availabilityZone": "nova",
INFO[0006]         "name": "hello",
INFO[0006]         "size": 1,
INFO[0006]         "status": "in-use",
INFO[0006]         "id": "ec41dcbb-9362-49ff-9650-7ab2e3ce7e5b",
INFO[0006]         "type": ""
INFO[0006]       }
INFO[0006]     }
INFO[0006]     0
INFO[0006]
DEBU[0006] mounting volume                               host=unix:///var/run/rexray/062151526.sock opts=&{OverwriteFS:false NewFSType: Preempt:true Opts:<nil>} time=1                              508406551683 volumeID=ec41dcbb-9362-49ff-9650-7ab2e3ce7e5b volumeName=
INFO[0006] mounting volume                               host=unix:///var/run/rexray/062151526.sock integrationDriver=linux opts=&{OverwriteFS:false NewFSType: Preempt                              :true Opts:<nil>} osDriver=linux service=cinder storageDriver=libstorage time=1508406551685 volumeID=ec41dcbb-9362-49ff-9650-7ab2e3ce7e5b volumeName=
DEBU[0006] xli localdevices success                      host=unix:///var/run/rexray/062151526.sock instanceID=cinder=49a7ddd2-4b78-4ad8-9c5d-93701db4f506 integrationD                              river=linux osDriver=linux server=helix-cockatoo-az service=cinder storageDriver=libstorage time=1508406551685 txCR=1508406551 txID=08fbc454-8719-4540-705b-c6169b4e84f                              9
INFO[0006]
INFO[0006]     -------------------------- HTTP REQUEST (CLIENT) -------------------------
INFO[0006]     GET /volumes/cinder/ec41dcbb-9362-49ff-9650-7ab2e3ce7e5b?attachments=7 HTTP/1.1
INFO[0006]     Host: libstorage-server
INFO[0006]     Libstorage-Instanceid: cinder=49a7ddd2-4b78-4ad8-9c5d-93701db4f506
INFO[0006]     Libstorage-Localdevices: cinder=/dev/dm-0::,/dev/dm-1::,/dev/vda::,/dev/vda1::,/dev/vdb::,/dev/vdc::,/dev/vdf::
INFO[0006]     Libstorage-Tx: txID=08fbc454-8719-4540-705b-c6169b4e84f9, txCR=1508406551
INFO[0006]
INFO[0006]
INFO[0006]     -------------------------- HTTP RESPONSE (CLIENT) -------------------------
INFO[0006]     HTTP/1.1 200 OK
INFO[0006]     Content-Length: 452
INFO[0006]     Content-Type: application/json
INFO[0006]     Date: Thu, 19 Oct 2017 09:49:11 GMT
INFO[0006]     Libstorage-Servername: helix-cockatoo-az
INFO[0006]
INFO[0006]     {
INFO[0006]       "attachments": [
INFO[0006]         {
INFO[0006]           "deviceName": "/dev/vdd",
INFO[0006]           "instanceID": {
INFO[0006]             "id": "49a7ddd2-4b78-4ad8-9c5d-93701db4f506",
INFO[0006]             "driver": "cinder",
INFO[0006]             "service": ""
INFO[0006]           },
INFO[0006]           "status": "",
INFO[0006]           "volumeID": "ec41dcbb-9362-49ff-9650-7ab2e3ce7e5b"
INFO[0006]         }
INFO[0006]       ],
INFO[0006]       "attachmentState": 2,
INFO[0006]       "availabilityZone": "nova",
INFO[0006]       "name": "hello",
INFO[0006]       "size": 1,
INFO[0006]       "status": "in-use",
INFO[0006]       "id": "ec41dcbb-9362-49ff-9650-7ab2e3ce7e5b",
INFO[0006]       "type": ""
INFO[0006]     }
INFO[0006]
INFO[0006]     -------------------------- HTTP REQUEST (CLIENT) -------------------------
INFO[0006]     GET /services/cinder?instance HTTP/1.1
INFO[0006]     Host: libstorage-server
INFO[0006]     Libstorage-Instanceid: cinder=49a7ddd2-4b78-4ad8-9c5d-93701db4f506
INFO[0006]     Libstorage-Tx: txID=971fdd49-f32c-4c0b-6c58-9c454e795206, txCR=1508406551
INFO[0006]
INFO[0006]
INFO[0006]     -------------------------- HTTP RESPONSE (CLIENT) -------------------------
INFO[0006]     HTTP/1.1 200 OK
INFO[0006]     Content-Length: 252
INFO[0006]     Content-Type: application/json
INFO[0006]     Date: Thu, 19 Oct 2017 09:49:11 GMT
INFO[0006]     Libstorage-Servername: helix-cockatoo-az
INFO[0006]
INFO[0006]     {
INFO[0006]       "name": "cinder",
INFO[0006]       "instance": {
INFO[0006]         "instanceID": {
INFO[0006]           "id": "49a7ddd2-4b78-4ad8-9c5d-93701db4f506",
INFO[0006]           "driver": "cinder",
INFO[0006]           "service": ""
INFO[0006]         },
INFO[0006]         "providerName": ""
INFO[0006]       },
INFO[0006]       "driver": {
INFO[0006]         "name": "cinder",
INFO[0006]         "type": "block"
INFO[0006]       }
INFO[0006]     }
DEBU[0006] built output template                         host=unix:///var/run/rexray/062151526.sock template={{define "printObject"}}{{printf "%v" .}}{{end}}{{define "                              printStringSlice"}}{{.}}{{end}}{{define "printPrettyJSON"}}{{. | jsonp}}{{end}}{{define "printScriptFields"}}{{.Path}}  {{.Installed}}  {{.Modified}}{{end}}{{define "p                              rintVolumeFields"}}{{.ID}}      {{.Name}}       {{.AttachmentState | printAttState}}    {{.Size}}{{end}}{{define "printMountFields"}}{{.ID}}    {{.Source}}     {{.Moun                              tPoint}}{{end}}{{define "printAuthTokenFields"}}{{.Subject}}    {{.IssuedAt}}   {{.NotBefore}}  {{.Expires}}{{end}}{{define "printVolumeID"}}{{.ID}}{{end}}{{define "pr                              intVolumeWithPathFields"}}{{.ID}}       {{.Name}}       {{.Volume.AttachmentState | printAttState}}     {{.Size}}       {{.Path}}{{end}}{{define "printSnapshotFields"}                              }{{.ID}}        {{.Name}}       {{.Status}}     {{.VolumeID}}{{end}}{{define "printInstanceFields"}}{{.InstanceID.ID}}  {{.Name}}       {{.ProviderName}}       {{.Regi                              on}}{{end}}{{define "printJSON"}}{{. | json}}{{end}}{{define "printEncVolumeFields"}}{{.ID}}    {{.Name}}       {{.AttachmentState | printAttState}}    {{.Size}}     {                              {.Encrypted}}{{end}}{{define "printEncVolumeWithPathFields"}}{{.ID}}    {{.Name}}       {{.Volume.AttachmentState | printAttState}}     {{.Size}}       {{.Encrypted}}{                              {.Path}}{{end}}{{define "printServiceFields"}}{{.Name}} {{.Driver.Name}}{{end}}ID       Name    Status  Size    Path
{{range sort .D "Name" }}{{template "printVolumeWithPathFields" .}}
{{end}} time=1508406551873
ID  Name  Status  Size  Path
FATA[0006] error mounting volume                         error=open /dev/vdd: no such file or directory volume=hello


root@node0:~# lsblk
NAME                MAJ:MIN RM SIZE RO TYPE MOUNTPOINT
vda                 253:0    0  20G  0 disk
└─vda1              253:1    0  20G  0 part /
vdb                 253:16   0  30G  0 disk
└─vgusers-lvusers   252:1    0  30G  0 lvm  /users
vdc                 253:32   0  70G  0 disk
└─vgdocker-lvdocker 252:0    0  70G  0 lvm  /var/lib/docker
vdf                 253:80   0   1G  0 disk
```

We can see the hello volume (1G) in the last ligne, device is vdf.
But Rexray tried to mount /dev/vdd.
I suppose a caching issue. 

I removed the line in the configuration
        #unmount:
        #  ignoreusedcount: true

The issue is the same.
I will try to disable the cache, or use cache in synchronous mode.

When I tried to mount without attach the volume, I get the same behavior : 

```
ID                                    Name   Status     Size
ec41dcbb-9362-49ff-9650-7ab2e3ce7e5b  hello  available  1
DEBU[0011] waiting until libStorage is stopped           host=unix:///var/run/rexray/062151526.sock time=1508407161754
DEBU[0011] done waiting on err chan; err chan is nil     host=unix:///var/run/rexray/062151526.sock time=1508407161754
DEBU[0011] completed cli execution                       time=1508407161754
INFO[0011] exiting process                               time=1508407161754
DEBU[0011] completed onExit at end of program            time=1508407161754



root@node0:~# lsblk
NAME                MAJ:MIN RM SIZE RO TYPE MOUNTPOINT
vda                 253:0    0  20G  0 disk
+-vda1              253:1    0  20G  0 part /
vdb                 253:16   0  30G  0 disk
+-vgusers-lvusers   252:1    0  30G  0 lvm  /users
vdc                 253:32   0  70G  0 disk
+-vgdocker-lvdocker 252:0    0  70G  0 lvm  /var/lib/docker



root@node0:~# rexray volume mount hello
INFO[0000] updated log level                             logLevel=debug
DEBU[0000] created scoped scope                          new=rexray.cli parentScopes=rexray,
DEBU[0000] os.args                                       time=1508407171570 val=[rexray volume mount hello]
DEBU[0000] activating libStorage                         cmd=mount time=1508407171570
DEBU[0000] parseSafeHost - no change                     postParse=unix:///var/run/rexray/062151526.sock preParse=unix:///var/run/rexray/062151526.sock time=1508407171578
DEBU[0000] read spec file                                host=unix:///var/run/rexray/062151526.sock path=/var/run/rexray/rexray.spec time=1508407171578
DEBU[0000] is local unix server active                   sock=/var/run/rexray/062151526.sock time=1508407171578
DEBU[0000] set host in context                           host=unix:///var/run/rexray/062151526.sock time=1508407171578
DEBU[0000] set host in config                            host=unix:///var/run/rexray/062151526.sock time=1508407171578
DEBU[0000] not starting embedded server; already running  host=unix:///var/run/rexray/062151526.sock time=1508407171578
DEBU[0000] creating libStorage client                    cmd=mount host=unix:///var/run/rexray/062151526.sock time=1508407171578
DEBU[0000] created scoped scope                          new=libstorage.client parentScopes=rexray.cli,rexray,
INFO[0000] configured logging                            host=unix:///var/run/rexray/062151526.sock libstorage.logging.httpRequests=true libstorage.logging.httpResponses=true libstorage.logging.level=debug time=1508407171686
INFO[0000] set client service name                       host=unix:///var/run/rexray/062151526.sock service=cinder serviceName=cinder time=1508407171691
DEBU[0000] got configured host address                   host=unix:///var/run/rexray/062151526.sock service=cinder storageDriver=libstorage time=1508407171697
DEBU[0000] disabling tls for unix sockets                host=unix:///var/run/rexray/062151526.sock service=cinder storageDriver=libstorage time=1508407171700
DEBU[0000] getHost proto == "unix"                       getHost=libstorage-server host=unix:///var/run/rexray/062151526.sock service=cinder storageDriver=libstorage time=1508407171701
INFO[0000] created libStorage client                     clientType=integration disableKeepAlive=false enableInstanceIDHeaders=true enableLocalDevicesHeaders=true host=unix:///var/run/rexray/062151526.sock lAddr=libstorage-server logRequests=true logResponses=true lsxPath= service=cinder storageDriver=libstorage time=1508407171723
INFO[0000] registered custom context key                 externalID=Libstorage-Tx internalID=2 keyBitmask=2
INFO[0000] registered custom context key                 externalID=Libstorage-Instanceid internalID=3 keyBitmask=2
INFO[0000] registered custom context key                 externalID=Libstorage-Localdevices internalID=4 keyBitmask=2
INFO[0000] registered custom context key                 externalID=Authorization internalID=5 keyBitmask=2
INFO[0000]
INFO[0000]     -------------------------- HTTP REQUEST (CLIENT) -------------------------
INFO[0000]     GET /services HTTP/1.1
INFO[0000]     Host: libstorage-server
INFO[0000]     Libstorage-Tx: txID=8278e0cc-4dd0-4089-4222-3fd5563d8e44, txCR=1508407171
INFO[0000]
DEBU[0000] successful connection                         host=unix:///var/run/rexray/062151526.sock service=cinder storageDriver=libstorage time=1508407171723
INFO[0000]
INFO[0000]     -------------------------- HTTP RESPONSE (CLIENT) -------------------------
INFO[0000]     HTTP/1.1 200 OK
INFO[0000]     Content-Length: 111
INFO[0000]     Content-Type: application/json
INFO[0000]     Date: Thu, 19 Oct 2017 09:59:31 GMT
INFO[0000]     Libstorage-Servername: helix-cockatoo-az
INFO[0000]
INFO[0000]     {
INFO[0000]       "cinder": {
INFO[0000]         "name": "cinder",
INFO[0000]         "driver": {
INFO[0000]           "name": "cinder",
INFO[0000]           "type": "block"
INFO[0000]         }
INFO[0000]       }
INFO[0000]     }
INFO[0000] initializing supported cache                  host=unix:///var/run/rexray/062151526.sock server=helix-cockatoo-az service=cinder storageDriver=libstorage time=1508407171726
DEBU[0000] cached supported flag                         host=unix:///var/run/rexray/062151526.sock server=helix-cockatoo-az service=cinder storageDriver=libstorage supported=15 time=1508407171726 txCR=1508407171 txID=f1f8547a-eebf-4648-6a8a-a84fa94a2b2f
INFO[0000] initializing instance ID cache                host=unix:///var/run/rexray/062151526.sock server=helix-cockatoo-az service=cinder storageDriver=libstorage time=1508407171726
DEBU[0005] cached instanceID                             host=unix:///var/run/rexray/062151526.sock instanceID=cinder=49a7ddd2-4b78-4ad8-9c5d-93701db4f506 server=helix-cockatoo-az service=cinder storageDriver=libstorage time=1508407176732 txCR=1508407171 txID=f568d6b1-4f46-4a11-6b82-52c65d87742f
DEBU[0005] xli instanceID success                        host=unix:///var/run/rexray/062151526.sock instanceID=cinder=49a7ddd2-4b78-4ad8-9c5d-93701db4f506 server=helix-cockatoo-az service=cinder storageDriver=libstorage time=1508407176732 txCR=1508407171 txID=f568d6b1-4f46-4a11-6b82-52c65d87742f
INFO[0005] successefully dialed libStorage server        host=unix:///var/run/rexray/062151526.sock server=helix-cockatoo-az service=cinder storageDriver=libstorage time=1508407176733
INFO[0005] storage driver initialized                    host=unix:///var/run/rexray/062151526.sock service=cinder storageDriver=libstorage time=1508407176733
INFO[0005] os driver initialized                         host=unix:///var/run/rexray/062151526.sock osDriver=linux service=cinder storageDriver=libstorage time=1508407176737
INFO[0005] linux integration driver successfully initialized  host=unix:///var/run/rexray/062151526.sock integrationDriver=linux libstorage.integration.volume.operations.create.default.IOPS=0 libstorage.integration.volume.operations.create.default.availabilityZone= libstorage.integration.volume.operations.create.default.fsType=ext4 libstorage.integration.volume.operations.create.default.size=1 libstorage.integration.volume.operations.create.default.type= libstorage.integration.volume.operations.create.implicit=true libstorage.integration.volume.operations.mount.path=/var/lib/rexray/volumes libstorage.integration.volume.operations.mount.rootPath=/data osDriver=linux service=cinder storageDriver=libstorage time=1508407176777
INFO[0005] path cache initializion disabled              host=unix:///var/run/rexray/062151526.sock integrationDriver=linux osDriver=linux service=cinder storageDriver=libstorage time=1508407176786
INFO[0005] libStorage integration driver successfully initialized  host=unix:///var/run/rexray/062151526.sock integrationDriver=linux libstorage.integration.volume.operations.create.disable=false libstorage.integration.volume.operations.mount.preempt=true libstorage.integration.volume.operations.path.cache.async=true libstorage.integration.volume.operations.path.cache.enabled=false libstorage.integration.volume.operations.remove.disable=true libstorage.integration.volume.operations.unmount.ignoreusedcount=false osDriver=linux service=cinder storageDriver=libstorage time=1508407176809
INFO[0005] integration driver initialized                host=unix:///var/run/rexray/062151526.sock integrationDriver=linux osDriver=linux service=cinder storageDriver=libstorage time=1508407176809
INFO[0005] created libStorage client                     host=unix:///var/run/rexray/062151526.sock integrationDriver=linux osDriver=linux service=cinder storageDriver=libstorage time=1508407176809
DEBU[0005] xli localdevices success                      host=unix:///var/run/rexray/062151526.sock instanceID=cinder=49a7ddd2-4b78-4ad8-9c5d-93701db4f506 integrationDriver=linux osDriver=linux server=helix-cockatoo-az service=cinder storageDriver=libstorage time=1508407176809 txCR=1508407176 txID=9f6fb197-a5ad-4689-54ee-61113091c2ad
INFO[0005]
INFO[0005]     -------------------------- HTTP REQUEST (CLIENT) -------------------------
INFO[0005]     GET /volumes/cinder?attachments=27 HTTP/1.1
INFO[0005]     Host: libstorage-server
INFO[0005]     Libstorage-Instanceid: cinder=49a7ddd2-4b78-4ad8-9c5d-93701db4f506
INFO[0005]     Libstorage-Localdevices: cinder=/dev/dm-0::,/dev/dm-1::,/dev/vda::,/dev/vda1::,/dev/vdb::,/dev/vdc::
INFO[0005]     Libstorage-Tx: txID=9f6fb197-a5ad-4689-54ee-61113091c2ad, txCR=1508407176
INFO[0005]
INFO[0006]
INFO[0006]     -------------------------- HTTP RESPONSE (CLIENT) -------------------------
INFO[0006]     HTTP/1.1 200 OK
INFO[0006]     Transfer-Encoding: chunked
INFO[0006]     Content-Type: application/json
INFO[0006]     Date: Thu, 19 Oct 2017 09:59:37 GMT
INFO[0006]     Libstorage-Servername: helix-cockatoo-az
INFO[0006]
INFO[0006]     a54
INFO[0006]     {
INFO[0006]       "1ed07588-3472-4a21-ac14-1a67709f5582": {
INFO[0006]         "attachments": [
INFO[0006]           {
INFO[0006]             "deviceName": "/dev/vda",
INFO[0006]             "instanceID": {
INFO[0006]               "id": "49a7ddd2-4b78-4ad8-9c5d-93701db4f506",
INFO[0006]               "driver": "cinder",
INFO[0006]               "service": ""
INFO[0006]             },
INFO[0006]             "status": "",
INFO[0006]             "volumeID": "1ed07588-3472-4a21-ac14-1a67709f5582"
INFO[0006]           }
INFO[0006]         ],
INFO[0006]         "attachmentState": 2,
INFO[0006]         "availabilityZone": "nova",
INFO[0006]         "name": "node0-boot",
INFO[0006]         "size": 20,
INFO[0006]         "status": "in-use",
INFO[0006]         "id": "1ed07588-3472-4a21-ac14-1a67709f5582",
INFO[0006]         "type": "svc-s21"
INFO[0006]       },
INFO[0006]       "3cd9680a-65ce-4bd8-a20b-e5d17439742a": {
INFO[0006]         "attachments": [
INFO[0006]           {
INFO[0006]             "deviceName": "/dev/vdc",
INFO[0006]             "instanceID": {
INFO[0006]               "id": "49a7ddd2-4b78-4ad8-9c5d-93701db4f506",
INFO[0006]               "driver": "cinder",
INFO[0006]               "service": ""
INFO[0006]             },
INFO[0006]             "status": "",
INFO[0006]             "volumeID": "3cd9680a-65ce-4bd8-a20b-e5d17439742a"
INFO[0006]           }
INFO[0006]         ],
INFO[0006]         "attachmentState": 2,
INFO[0006]         "availabilityZone": "nova",
INFO[0006]         "name": "node0-var-lib-docker",
INFO[0006]         "size": 70,
INFO[0006]         "status": "in-use",
INFO[0006]         "id": "3cd9680a-65ce-4bd8-a20b-e5d17439742a",
INFO[0006]         "type": "svc-s21"
INFO[0006]       },
INFO[0006]       "98d91360-4310-49f4-8c7d-15f6b5e450d0": {
INFO[0006]         "attachmentState": 3,
INFO[0006]         "availabilityZone": "nova",
INFO[0006]         "name": "yvao4270-users",
INFO[0006]         "size": 30,
INFO[0006]         "status": "available",
INFO[0006]         "id": "98d91360-4310-49f4-8c7d-15f6b5e450d0",
INFO[0006]         "type": "svc-s21"
INFO[0006]       },
INFO[0006]       "b1c1663f-2eb5-4a87-86e8-bf47a9a08f1f": {
INFO[0006]         "attachmentState": 3,
INFO[0006]         "availabilityZone": "nova",
INFO[0006]         "name": "",
INFO[0006]         "size": 5,
INFO[0006]         "status": "available",
INFO[0006]         "id": "b1c1663f-2eb5-4a87-86e8-bf47a9a08f1f",
INFO[0006]         "type": "svc-s01"
INFO[0006]       },
INFO[0006]       "b9603f10-c5d5-4b0a-a9a0-47936be39074": {
INFO[0006]         "attachments": [
INFO[0006]           {
INFO[0006]             "deviceName": "/dev/vdb",
INFO[0006]             "instanceID": {
INFO[0006]               "id": "49a7ddd2-4b78-4ad8-9c5d-93701db4f506",
INFO[0006]               "driver": "cinder",
INFO[0006]               "service": ""
INFO[0006]             },
INFO[0006]             "status": "",
INFO[0006]             "volumeID": "b9603f10-c5d5-4b0a-a9a0-47936be39074"
INFO[0006]           }
INFO[0006]         ],
INFO[0006]         "attachmentState": 2,
INFO[0006]         "availabilityZone": "nova",
INFO[0006]         "name": "node0-users",
INFO[0006]         "size": 30,
INFO[0006]         "status": "in-use",
INFO[0006]         "id": "b9603f10-c5d5-4b0a-a9a0-47936be39074",
INFO[0006]         "type": "svc-s21"
INFO[0006]       },
INFO[0006]       "db5d7770-2f62-4400-9808-0554c48c446f": {
INFO[0006]         "attachmentState": 3,
INFO[0006]         "availabilityZone": "nova",
INFO[0006]         "name": "yvao4270-boot",
INFO[0006]         "size": 20,
INFO[0006]         "status": "available",
INFO[0006]         "id": "db5d7770-2f62-4400-9808-0554c48c446f",
INFO[0006]         "type": "svc-s21"
INFO[0006]       },
INFO[0006]       "ec41dcbb-9362-49ff-9650-7ab2e3ce7e5b": {
INFO[0006]         "attachmentState": 3,
INFO[0006]         "availabilityZone": "nova",
INFO[0006]         "name": "hello",
INFO[0006]         "size": 1,
INFO[0006]         "status": "available",
INFO[0006]         "id": "ec41dcbb-9362-49ff-9650-7ab2e3ce7e5b",
INFO[0006]         "type": ""
INFO[0006]       }
INFO[0006]     }
INFO[0006]     0
INFO[0006]
DEBU[0006] mounting volume                               host=unix:///var/run/rexray/062151526.sock opts=&{OverwriteFS:false NewFSType: Preempt:true Opts:<nil>} time=1508407177743 volumeID=ec41dcbb-9362-49ff-9650-7ab2e3ce7e5b volumeName=
INFO[0006] mounting volume                               host=unix:///var/run/rexray/062151526.sock integrationDriver=linux opts=&{OverwriteFS:false NewFSType: Preempt:true Opts:<nil>} osDriver=linux service=cinder storageDriver=libstorage time=1508407177743 volumeID=ec41dcbb-9362-49ff-9650-7ab2e3ce7e5b volumeName=
DEBU[0006] xli localdevices success                      host=unix:///var/run/rexray/062151526.sock instanceID=cinder=49a7ddd2-4b78-4ad8-9c5d-93701db4f506 integrationDriver=linux osDriver=linux server=helix-cockatoo-az service=cinder storageDriver=libstorage time=1508407177744 txCR=1508407177 txID=0d63cd33-d19d-4718-61ec-98c80c7e2050
INFO[0006]
INFO[0006]     -------------------------- HTTP REQUEST (CLIENT) -------------------------
INFO[0006]     GET /volumes/cinder/ec41dcbb-9362-49ff-9650-7ab2e3ce7e5b?attachments=7 HTTP/1.1
INFO[0006]     Host: libstorage-server
INFO[0006]     Libstorage-Instanceid: cinder=49a7ddd2-4b78-4ad8-9c5d-93701db4f506
INFO[0006]     Libstorage-Localdevices: cinder=/dev/dm-0::,/dev/dm-1::,/dev/vda::,/dev/vda1::,/dev/vdb::,/dev/vdc::
INFO[0006]     Libstorage-Tx: txID=0d63cd33-d19d-4718-61ec-98c80c7e2050, txCR=1508407177
INFO[0006]
INFO[0006]
INFO[0006]     -------------------------- HTTP RESPONSE (CLIENT) -------------------------
INFO[0006]     HTTP/1.1 200 OK
INFO[0006]     Content-Length: 175
INFO[0006]     Content-Type: application/json
INFO[0006]     Date: Thu, 19 Oct 2017 09:59:37 GMT
INFO[0006]     Libstorage-Servername: helix-cockatoo-az
INFO[0006]
INFO[0006]     {
INFO[0006]       "attachmentState": 3,
INFO[0006]       "availabilityZone": "nova",
INFO[0006]       "name": "hello",
INFO[0006]       "size": 1,
INFO[0006]       "status": "available",
INFO[0006]       "id": "ec41dcbb-9362-49ff-9650-7ab2e3ce7e5b",
INFO[0006]       "type": ""
INFO[0006]     }
DEBU[0006] performing precautionary unmount              host=unix:///var/run/rexray/062151526.sock integrationDriver=linux osDriver=linux service=cinder storageDriver=libstorage time=1508407177907
DEBU[0006] xli localdevices success                      host=unix:///var/run/rexray/062151526.sock instanceID=cinder=49a7ddd2-4b78-4ad8-9c5d-93701db4f506 integrationDriver=linux osDriver=linux server=helix-cockatoo-az service=cinder storageDriver=libstorage time=1508407177908 txCR=1508407177 txID=f184e124-f806-463b-69a0-efa48cfdb0b8
INFO[0006]
INFO[0006]     -------------------------- HTTP REQUEST (CLIENT) -------------------------
INFO[0006]     POST /volumes/cinder/ec41dcbb-9362-49ff-9650-7ab2e3ce7e5b?attach HTTP/1.1
INFO[0006]     Host: libstorage-server
INFO[0006]     Libstorage-Instanceid: cinder=49a7ddd2-4b78-4ad8-9c5d-93701db4f506
INFO[0006]     Libstorage-Localdevices: cinder=/dev/dm-0::,/dev/dm-1::,/dev/vda::,/dev/vda1::,/dev/vdb::,/dev/vdc::
INFO[0006]     Libstorage-Tx: txID=f184e124-f806-463b-69a0-efa48cfdb0b8, txCR=1508407177
INFO[0006]
INFO[0006]     {"force":true}
INFO[0012]
INFO[0012]     -------------------------- HTTP RESPONSE (CLIENT) -------------------------
INFO[0012]     HTTP/1.1 200 OK
INFO[0012]     Content-Length: 537
INFO[0012]     Content-Type: application/json
INFO[0012]     Date: Thu, 19 Oct 2017 09:59:44 GMT
INFO[0012]     Libstorage-Servername: helix-cockatoo-az
INFO[0012]
INFO[0012]     {
INFO[0012]       "volume": {
INFO[0012]         "attachments": [
INFO[0012]           {
INFO[0012]             "deviceName": "/dev/vdd",
INFO[0012]             "instanceID": {
INFO[0012]               "id": "49a7ddd2-4b78-4ad8-9c5d-93701db4f506",
INFO[0012]               "driver": "cinder",
INFO[0012]               "service": ""
INFO[0012]             },
INFO[0012]             "status": "",
INFO[0012]             "volumeID": "ec41dcbb-9362-49ff-9650-7ab2e3ce7e5b"
INFO[0012]           }
INFO[0012]         ],
INFO[0012]         "attachmentState": 2,
INFO[0012]         "availabilityZone": "nova",
INFO[0012]         "name": "hello",
INFO[0012]         "size": 1,
INFO[0012]         "status": "in-use",
INFO[0012]         "id": "ec41dcbb-9362-49ff-9650-7ab2e3ce7e5b",
INFO[0012]         "type": ""
INFO[0012]       },
INFO[0012]       "attachToken": "/dev/vdd"
INFO[0012]     }
DEBU[0042] built output template                         host=unix:///var/run/rexray/062151526.sock template={{define "printVolumeWithPathFields"}}{{.ID}}      {{.Name}}       {{.Volume.AttachmentState | printAttState}}  {{.Size}}       {{.Path}}{{end}}{{define "printInstanceFields"}}{{.InstanceID.ID}}      {{.Name}}       {{.ProviderName}}       {{.Region}}{{end}}{{define "printMountFields"}}{{.ID}}       {{.Source}}     {{.MountPoint}}{{end}}{{define "printStringSlice"}}{{.}}{{end}}{{define "printVolumeFields"}}{{.ID}}    {{.Name}}       {{.AttachmentState | printAttState}}    {{.Size}}{{end}}{{define "printEncVolumeWithPathFields"}}{{.ID}}     {{.Name}}       {{.Volume.AttachmentState | printAttState}}     {{.Size}}       {{.Encrypted}}  {{.Path}}{{end}}{{define "printAuthTokenFields"}}{{.Subject}}        {{.IssuedAt}}   {{.NotBefore}}  {{.Expires}}{{end}}{{define "printObject"}}{{printf "%v" .}}{{end}}{{define "printVolumeID"}}{{.ID}}{{end}}{{define "printEncVolumeFields"}}{{.ID}}  {{.Name}}       {{.AttachmentState | printAttState}}    {{.Size}}       {{.Encrypted}}{{end}}{{define "printSnapshotFields"}}{{.ID}}    {{.Name}}       {{.Status}}     {{.VolumeID}}{{end}}{{define "printServiceFields"}}{{.Name}} {{.Driver.Name}}{{end}}{{define "printPrettyJSON"}}{{. | jsonp}}{{end}}{{define "printScriptFields"}}{{.Path}}  {{.Installed}}  {{.Modified}}{{end}}{{define "printJSON"}}{{. | json}}{{end}}ID      Name    Status  Size    Path
{{range sort .D "Name" }}{{template "printVolumeWithPathFields" .}}
{{end}} time=1508407214333
ID  Name  Status  Size  Path
FATA[0042] error mounting volume                         volume=hello



root@node0:~# lsblk
NAME                MAJ:MIN RM SIZE RO TYPE MOUNTPOINT
vda                 253:0    0  20G  0 disk
+-vda1              253:1    0  20G  0 part /
vdb                 253:16   0  30G  0 disk
+-vgusers-lvusers   252:1    0  30G  0 lvm  /users
vdc                 253:32   0  70G  0 disk
+-vgdocker-lvdocker 252:0    0  70G  0 lvm  /var/lib/docker
vdf                 253:80   0   1G  0 disk
```

Again, Rexray use the device name provided by openstack cinder ("attachToken": "/dev/vdd") instead of the real device on the node (/dev/vdf).

**Comment by olivibernard** [comment]
When I disable the volume path cache, same problem
```
libstorage:
  integration:
    volume:
      operations:
        path:
          cache:
            enabled: false
```

**Comment by olivibernard** [comment]
The device issue is a well known problem : 
https://ask.openstack.org/en/question/81702/nova-volume-attach-with-device-option-not-honoring-given-device-name/

Kubernetes project merged a PR to avoid relying on device name provided by Cinder (#33270)
https://github.com/kubernetes/kubernetes/issues/33128

**Comment by clintkitson** [comment]
Thanks for pointing this one out.

# Our return of what Cinder says
https://github.com/codedellemc/rexray/blob/master/libstorage/drivers/storage/cinder/storage/cinder_storage.go#L627-L640

# Our executor using /proc/partitions - I don't believe this is actually used anywhere. It should be /dev/disk/by-id and the results should be used when we wait for volume and then return after attach.
https://github.com/codedellemc/rexray/blob/master/libstorage/drivers/storage/cinder/executor/cinder_executor.go#L203

# The PR files
https://github.com/hpcloud/kubernetes/pull/16/files

# The logic to select disks form /dev/disk
https://github.com/hpcloud/kubernetes/blob/41fa2111b585df72a7726eeedf386abf6bf85fb1/pkg/cloudprovider/providers/openstack/openstack.go#L666-L688

# The configurable option
https://github.com/hpcloud/kubernetes/blob/41fa2111b585df72a7726eeedf386abf6bf85fb1/pkg/cloudprovider/providers/openstack/openstack.go#L740-L743

**Comment by clintkitson** [comment]
@MatMaul Any comment on this one?

**Comment by MatMaul** [comment]
Indeed known problem with Cinder/Nova reporting wrong device in some configuration, I never had the problem. We could switch to the k8s heuristic.
Note that we had a report recently of someone using k8s with HyperV and the heuristic doesn't exist for it, we should fallback on the reported device if the heuristic fails.

**Comment by MatMaul** [comment]
I quickly had a look, I think we have a problem doing that with the current design.
LocalDevices wants a map of path->volumeId.
I don't think I can easily access this information, I only have access to the first 20 chars of the volume id.

The current code just reports every devices without volume ID and we use the one reported by the API. Which is wrong in this case.

Ideas ? pass the volume ID(s) requested in a new method, something like ```MatchingDevices(string[] volumeIds) map[string]string``` ?

**Comment by MatMaul** [comment]
@akutz ;)

**Comment by clintkitson** [comment]
Here is an example of getting the local devices for a different prefix
https://github.com/codedellemc/rexray/blob/master/libstorage/drivers/storage/scaleio/executor/scaleio_executor.go#L68-L81

Then using those devices to populate the attachment info on the volumes
https://github.com/codedellemc/rexray/blob/master/libstorage/drivers/storage/scaleio/storage/scaleio_storage.go#L170-L172

Then returning after an attach a volume with attachment info including that actual device name
https://github.com/codedellemc/rexray/blob/master/libstorage/drivers/storage/scaleio/storage/scaleio_storage.go#L514-L523

**Comment by olivibernard** [comment]
Hi there !
Any update on this issue ?

**Comment by fabio-barile** [comment]
Hi
Do you have a plan/date for delivering a fix for this issue ?
We often meet this issue in our internal cloud based on openstack.
BR

**Comment by clintkitson** [comment]
We would love to see a community contribution to get this updated. In fact there is another issue for OpenStack that can also use some attention https://github.com/thecodeteam/rexray/issues/1113.

The team has been busy making some pretty big shifts the CSI implementations. This would actually be a great time to think about doing a `csi-openstack` driver with these updates.

**Comment by dave08** [comment]
Is there any workaround for this issue? I'm having the same problem with the Digital Ocean driver (they're using Cinder?)...
